nucypher/blockchain/eth/actors.py

```diff
@@ -410,176 +432,102 @@ def total_staked(self) -> NU:
-            raise self.MinerError(f'Cannot dive an expired stake')
-
-        # Validate stake division parameters
+        # Select stake to divide
+        try:
+            current_stake = self.stakes[stake_index]
+        except KeyError:
+            if len(self.stakes):
+                message = f"Cannot divide stake - No stake exists with index {stake_index}."
+            else:
+                message = "Cannot divide stake - There are no active stakes."
+            raise Stake.StakingError(message)
+
+        # Calculate stake duration in periods
         if expiration:
             additional_periods = datetime_to_period(datetime=expiration) - current_stake.end_period
             if additional_periods <= 0:
-                raise self.MinerError("Expiration {} must be at least 1 period from now.".format(expiration))
-
-        if target_value >= current_stake.value:
-            raise self.MinerError(f"Cannot divide stake; Value ({target_value}) must be less "
-                                  f"than the existing stake value {current_stake.value}.")
-
-        #
-        # Generate Stakes
-        #
-
-        # Modified Original Stake
-        new_stake_1 = Stake(owner_address=self.checksum_public_address,
-                            index=len(self.stakes)+1,
-                            start_period=current_stake.start_period,
-                            end_period=current_stake.end_period,
-                            value=target_value,
-                            economics=self.economics)
-
-        # New Derived Stake
-        new_stake_2 = Stake(owner_address=self.checksum_public_address,
-                            index=len(self.stakes)+1,
-                            start_period=current_stake.start_period,
-                            end_period=current_stake.end_period + additional_periods,
-                            value=current_stake.value - target_value,
-                            economics=self.economics)
-
-        # Ensure both halves are for valid amounts
-        new_stake_1.validate_value()
-        new_stake_2.validate_value()
-
-        # Transmit the stake division transaction
-        tx = self.miner_agent.divide_stake(miner_address=self.checksum_public_address,
-                                           stake_index=stake_index,
-                                           target_value=int(target_value),
-                                           periods=additional_periods)
-        self.blockchain.wait_for_receipt(tx)
+                raise Stake.StakingError("Expiration {} must be at least 1 period from now.".format(expiration))
 
-        # Update stake cache
-        self.__read_stakes()
+        # Check the new stake will not exceed the staking limit
+        if (self.total_staked + target_value) > self.economics.maximum_allowed_locked:
+            raise Stake.StakingError(f"Cannot divide stake - Maximum stake value exceeded with a target value of {target_value}.")
```

Comment:
My mistake in previous review, this part should be in `initialize_stake`. In `divide_stake` `total_staked` remains the same

Reply:
:heavy_check_mark: